cmd/provider/purge.go

```diff
@@ -0,0 +1,43 @@
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cloudquery/cloudquery/cmd/utils"
+	"github.com/cloudquery/cloudquery/pkg/errors"
+	"github.com/cloudquery/cloudquery/pkg/ui/console"
+	"github.com/spf13/cobra"
+)
+
+var (
+	lastUpdate time.Duration
+	dryRun     bool
+	purgeShort = "Remove stale resources from one or more providers in database"
+)
+
+func newCmdProviderPurge() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "purge [provider]",
+		Short: purgeShort,
+		Long:  purgeShort,
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := console.CreateClient(cmd.Context(), utils.GetConfigFile(), false, nil, utils.InstanceId)
+			if err != nil {
+				return err
+			}
+			diags := c.RemoveStaleData(cmd.Context(), lastUpdate, dryRun, args)
+			errors.CaptureDiagnostics(diags, map[string]string{"command": "provider_purge"})
+			if diags.HasErrors() {
+				return fmt.Errorf("failed to remove stale data")
```

Comment:
I know you didn't write this code, but if I am already looking at it, I think we should be consistent with our error value returned. Either with the `%w` or return the where more diags are in the logs.
```suggestion
			return fmt.Errorf("failed to remove stale data: %w", diags)
```
```suggestion
				return fmt.Errorf("failed to remove stale data")
```